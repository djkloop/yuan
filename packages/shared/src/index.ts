export * from "./base"
export * from "./install"
export * from "./tools"
