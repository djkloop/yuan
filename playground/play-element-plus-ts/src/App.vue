<template>
  <YuanBasicLayout>
    <RouterView />
  </YuanBasicLayout>
</template>

<script setup lang="ts">
import YuanBasicLayout from '@/components/YuanLayout/YuanLayout';

</script>
<style lang="scss">
html,
body {
  height: 100%;
  margin: 0;
  padding: 0;
}

#app {
  height: 100%;
}
</style>
