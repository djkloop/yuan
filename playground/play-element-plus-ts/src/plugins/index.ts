import type { App } from 'vue';

import ElementPlusPlugin from './element-plus';
import VueTippyPlugin from './vue-tippy';
import VueRouterPlugin from './vue-router';
import PiniaPlugin from "./pinia";

const YuanPlugins = {
  use(app: App): App {
    app.use(PiniaPlugin)
    app.use(VueRouterPlugin)
    app.use(ElementPlusPlugin)
    app.use(VueTippyPlugin)
    return app
  },
}

export default YuanPlugins;
