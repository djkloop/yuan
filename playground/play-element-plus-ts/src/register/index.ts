/// 开始注册全局事件/数据
import { App } from "vue";
import StoreRegister from "./store"
let resgiterInstance: Register | null = null
class Register {

  constructor(app?: App) {
    this.initStore()
  }

  initStore() {
    // 注册全局数据
    StoreRegister.initGlobalStore()
  }

  static init(app?: App): Register {
    if (!resgiterInstance) {
      resgiterInstance = new Register(app)
    }
    return resgiterInstance
  }
}


export default Register
