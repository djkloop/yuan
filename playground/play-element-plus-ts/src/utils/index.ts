export function uuid() {
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
    var r = Math.random() * 16 | 0,
      v = c == 'x' ? r : (r & 0x3 | 0x8);
    return v.toString(16);
  });
}

// 动态加载组件
export const loadScript = (src: string) => {
  return new Promise((resolve, reject) => {
    const sc = document.createElement('script')
    sc.setAttribute('type', 'module')
    sc.setAttribute('id', src.replace('/', ''))
    function _onLoad() {
      resolve(src)
      sc.onload = sc.onerror = null
    }
    sc.onload = _onLoad;
    sc.onerror = reject;
    sc.setAttribute('src', '.' + src)
    sc.crossOrigin = "anonymous"
    document.head.append(sc)
  })
}
