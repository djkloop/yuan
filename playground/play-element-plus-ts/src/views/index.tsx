import { defineComponent } from "vue";
import { useRouter } from "vue-router"

const Home = defineComponent({
  name: "YuanHome",
  setup() {
    return () => {
      const router = useRouter()
      router.push('/editor')
      return (
        <div>
          Home.tsx
        </div>
      )
    }
  }
})

export default Home
