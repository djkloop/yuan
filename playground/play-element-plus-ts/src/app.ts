import { IElement, IProject, IPage } from 'yuan-types'
import { uuid } from "./utils"

export class Project implements IProject {
  name: string = "New Project";
  type: string = "web";
  description: string = "New Project Description";
  pages: Page[] = [];

  constructor() {
  }

  // 
  static create(uProject?: IProject) {
    // 创建一个项目
    const project = new Project();
    if (uProject) {
      project.name = uProject.name
      project.description = uProject.description
      project.pages = uProject.pages.map(up => Page.create(up))
    } else {
      // 创建一个页面
      const page = Page.create();
      // 添加进当前项目
      project.addPage(page)
    }
    return project
  }


  // 添加页面
  addPage(page: Page) {
    this.pages.push(page)
  }

  // 删除页面
  removePage(page: Page) {
    const idx = this.pages.findIndex(p => p.id === page.id)
    if (idx !== -1) {
      this.pages.splice(idx, 1)
    }
  }

  // 插入页面
  insertPage(index: number, page: Page) {
    this.pages.splice(index, 0, page)
  }

  // 获取page json
  getPageJson() {
    return {
      name: this.name,
      description: this.description,
      pages: this.pages.map(page => page.getPageJson())
    }
  }
}

// 项目中的每个页面类
export class Page implements IPage {

  id: string = "PageID"
  name: string = "New Page";
  description: string = "New Page Description";
  elements: PageElement[] = [];

  constructor() {

  }

  static create(uPage?: IPage) {
    const page = new Page()
    if (uPage) {
      page.description = uPage.description
      // 
      page.id = uPage.id
      page.name = uPage.name
      page.elements = uPage.elements.map(element => PageElement.create(element))
    }
    return page
  }

  // 添加一个元素
  addElement(element: PageElement) {
    this.elements.push(element)
  }

  // 删除一个元素
  removeElement(element: PageElement) {
    const idx = this.elements.findIndex(e => e.id === element.id)
    if (idx !== -1) {
      this.elements.splice(idx, 1)
    }
  }

  // 插入一个元素
  insertElement(index: number, element: PageElement) {
    this.elements.splice(index, 0, element)
  }

  public getPageJson() {
    return {
      id: this.id,
      name: this.name,
      description: this.description,
      elements: this.elements.map(element => element.getElementJson())
    }
  }

}

// 页面上的每个元素类
export class PageElement implements IElement {
  id: string = uuid();
  name: string = "Element Name";
  mId: number = 0;
  mVersion: string = "Null Version";
  constructor() { }

  static create(uElement?: IElement) {
    const element = new PageElement()
    if (uElement) {
      element.id = uElement.id
      element.name = uElement.name
      element.mId = uElement?.mId
      element.mVersion = uElement?.mVersion
    }
    return element
  }

  getElementJson() {
    return {
      id: this.id,
      name: this.name,
      mId: this.mId,
      mVersion: this.mVersion
    }
  }
}
