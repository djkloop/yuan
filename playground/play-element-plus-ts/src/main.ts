import "uno.css"
import { createApp, App as VueApp } from 'vue'
import * as Vue from "vue"
import YuanPlugins from "./plugins"
import YuanRegister from "./register"
import { loadScript } from "./utils"
import App from './App.vue'
import { IMaterial } from "yuan-types"
import { materialList } from "@/data"


// 全局注册Vue
window['Vue'] = Vue


// YuanLoadMicroComponents.load()

function runApp(app: VueApp) {
  YuanPlugins.use(app)
  YuanRegister.init()
  app.mount('#app')
}

const mSources = materialList.map(m => m.source)
Promise.all(mSources.map(ms => loadScript(ms))).then(() => {
  const app = createApp(App)
  materialList.forEach(m => {
    if (window.YuanMicroComponents) {
      const { render } = window.YuanMicroComponents[m.name]
      app.component(m.name, render)
    }
  })
  runApp(app)
})
