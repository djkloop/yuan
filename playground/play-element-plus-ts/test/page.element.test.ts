import { describe, expect, test } from 'vitest'
import { Project } from "../src/app"

describe('page.element', () => {
  const project = Project.create();
  test('project.name', () => {
    expect(project.name).toEqual('New Project')
  })
  test('project.pages.length === 1', () => {
    expect(project.pages.length).toBe(1)
  })

  test('project.json', () => {
    expect(project.getPageJson()).toMatchSnapshot()
  })

})
