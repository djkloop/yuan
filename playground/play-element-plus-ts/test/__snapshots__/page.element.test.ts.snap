// Vitest Snapshot v1

exports[`page.element > project.json 1`] = `
{
  "description": "New Project Description",
  "name": "New Project",
  "pages": [
    {
      "description": "New Page Description",
      "elements": [],
      "id": "PageID",
      "name": "New Page",
    },
  ],
}
`;
